client/controllers/events/eventFiltration.coffee

```diff
@@ -60,3 +119,24 @@ Template.eventFiltration.events
     instance.$('.other-properties input:checked').each (i, input) ->
       otherProps[instance.removePropPrefix(input.id)] = true
     instance.properties.set(otherProps)
+
+  'change .locations select': (event, instance) ->
+    instance.countryLevel.set(event.target.value)
+
+  'click .location-list li': (event, instance) ->
+    selectedLocations = instance.selectedLocations
+    if selectedLocations.findOne(id: @_id)
```

Comment:
I ran into an issue with duplicate ids (I think this is the issue I had in the past). Any way around it?